docs/data/charts/areas-demo/AreaChartFillByValue.js

```diff
@@ -0,0 +1,165 @@
+import * as React from 'react';
```

Comment:
I like a lot the idea of having two colors for positive and for minimal values, it's a super easy to understand use case for a color that depends on the value.

Did you consider using dates for the X data instead of Pages?
This would probably be closer from a real use case.

A super small dataset with the average temperature month by month of a country with both positive and negative could make sense here.
For instance in Montreal

![image](https://github.com/mui/mui-x/assets/3309670/447c6cef-ff0e-4b7f-8e8a-bf112950b36d)